subsys/net/lib/mqtt_helper/mqtt_helper.c

```diff
@@ -150,40 +153,79 @@ static int certificates_provision(void)
 		return 0;
 	}
 
-	err = tls_credential_add(CONFIG_AZURE_IOT_HUB_SEC_TAG,
-				 TLS_CREDENTIAL_CA_CERTIFICATE,
-				 ca_certificate,
-				 sizeof(ca_certificate));
-	if (err < 0) {
-		LOG_ERR("Failed to register CA certificate: %d",
-			err);
-		return err;
+	if (sizeof(ca_certificate) > 0) {
+		err = tls_credential_add(CONFIG_MQTT_HELPER_SEC_TAG,
+					 TLS_CREDENTIAL_CA_CERTIFICATE,
+					 ca_certificate,
+					 sizeof(ca_certificate));
+		if (err < 0) {
+			LOG_ERR("Failed to register CA certificate: %d", err);
+			return err;
+		}
 	}
 
-	err = tls_credential_add(CONFIG_AZURE_IOT_HUB_SEC_TAG,
-				 TLS_CREDENTIAL_PRIVATE_KEY,
-				 private_key,
-				 sizeof(private_key));
-	if (err < 0) {
-		LOG_ERR("Failed to register private key: %d", err);
-		return err;
+	if (sizeof(private_key) > 0) {
+		err = tls_credential_add(CONFIG_MQTT_HELPER_SEC_TAG,
+					 TLS_CREDENTIAL_PRIVATE_KEY,
+					 private_key,
+					 sizeof(private_key));
+		if (err < 0) {
+			LOG_ERR("Failed to register private key: %d", err);
+			return err;
+		}
 	}
 
-	err = tls_credential_add(CONFIG_AZURE_IOT_HUB_SEC_TAG,
-				 TLS_CREDENTIAL_SERVER_CERTIFICATE,
-				 device_certificate,
-				 sizeof(device_certificate));
-	if (err < 0) {
-		LOG_ERR("Failed to register public certificate: %d",
-			err);
-		return err;
+	if (sizeof(device_certificate) > 0) {
+		err = tls_credential_add(CONFIG_MQTT_HELPER_SEC_TAG,
+					 TLS_CREDENTIAL_SERVER_CERTIFICATE,
+					 device_certificate,
+					 sizeof(device_certificate));
+		if (err < 0) {
+			LOG_ERR("Failed to register public certificate: %d", err);
+			return err;
+		}
+	}
+
+	/* Secondary security tag entries. */
+
+	if ((sizeof(ca_certificate) > 0) && (CONFIG_MQTT_HELPER_SECONDARY_SEC_TAG != -1)) {
```

Comment:
Looks like we're passing in the same certs and key twice here. I guess we need to have two sets, one for primary and one for secondary tag

Reply:
Fixed